pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go

```diff
@@ -281,3 +281,106 @@ func TestPipelineRunGetServiceAccountName(t *testing.T) {
 		}
 	}
 }
+
+func TestPipelineRunGetPodSpecSABackcompatibility(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		pr      *v1alpha1.PipelineRun
+		saNames map[string]string
```

Comment:
minor nit: Can we be clearer that `saNames` is the expected/wanted output from the tests...maybe call the field `expectedSAs` or add a comment saying that this is a map of wanted SA names for each task name?